plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,798 @@
+        else:
+            license_file_name = f"{module.params['license_id']}_iDRAC_license.xml"
+        payload["FileName"] = license_file_name
+        license_status = self.idrac.invoke_request(export_license_url, "POST", data=payload)
+        return license_status
+
+
+class ImportLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        self.__check_file_extension(module)
+        if share_type == "local":
+            import_license_status = self.__import_license_local(module)
+        elif share_type in ["http", "https"]:
+            import_license_status = self.__import_license_http(module)
+            job_status = get_job_status(self, module, import_license_status)
+        elif share_type == "cifs":
+            import_license_status = self.__import_license_cifs(module)
+            job_status = get_job_status(self, module, import_license_status)
+        elif share_type == "nfs":
+            import_license_status = self.__import_license_nfs(module)
+            job_status = get_job_status(self, module, import_license_status)
+        status = import_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_IMPORT_MSG, changed=True, job_details=job_status)
+            else:
+                module.exit_json(msg=FAILURE_IMPORT_MSG, failed=True, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_IMPORT_MSG, changed=True, job_details={})
+            else:
+                module.exit_json(msg=FAILURE_IMPORT_MSG, failed=True, job_details={})
+        return import_license_status
+
+    def __import_license_local(self, module):
+        uri = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, SYSTEMS_URI)
+        import_license_url = f"{uri[0]}/{IMPORT_LICENSE_LOCAL}"
+        resource_id = uri[0].split("/")[-1]
+        payload = {}
+        path = module.params.get('share_parameters').get('share_name')
+        if not (os.path.exists(path) or os.path.isdir(path)):
+            module.exit_json(msg=f"Provided directory path '{path}' is not valid.", failed=True)
+        file_path = module.params.get('share_parameters').get('share_name') + "/" + module.params.get('share_parameters').get('file_name')
+        file_exits = os.path.exists(file_path)
+        if file_exits:
+            with open(file_path, "rb") as cert:
+                cert_content = cert.read()
+                read_file = base64.encodebytes(cert_content).decode('ascii')
+        else:
+            module.exit_json(msg=NO_FILE_MSG, failed=True)
+        payload["LicenseFile"] = read_file
+        payload["FQDD"] = resource_id
```

Comment:
Can't we take common parameters (FQDD, ImportOptions) required for import (HTTP/CIFS/NFS) outside and call it. Will reduce the code for calling it repeatedly.

Reply:
No changes done. Will proceed with existing approach